File: 강의 자료/02-알고리즘/09~11-그래프 탐색/문제별 코드/2606-바이러스/solution.py
import sys
from collections import deque
si = sys.stdin.readline
n = int(si())
m = int(si())
adj = [[] for _ in range(n + 1)]

for _ in range(m):
    x, y = list(map(int, si().split()))
    adj[x].append(y)
    adj[y].append(x)

visit = [0] * (n + 1)
def bfs(x):
    queue = deque()
    queue.append(x)
    visit[x] = 1
    while queue:
        x = queue.popleft()
        for y in adj[x]:
            if visit[y]:
                continue
            visit[y] = 1
            queue.append(y)

bfs(1)
print(sum(visit) - 1)

File: 강의 자료/02-알고리즘/09~11-그래프 탐색/문제별 코드/2606-바이러스/solution.cpp
#include <iostream>
#include <algorithm>
#pragma warning(disable:4996)
using namespace std;

#include <vector>
vector<int> adj[105];
int n, m;
void input() {
	cin >> n >> m;
	for (int i = 1; i <= m; i++) {
		int u, v;
		cin >> u >> v;
		adj[u].push_back(v);
		adj[v].push_back(u);
	}
}

bool visit[105];
#include <queue>
void BFS(int S) {
	for (int i = 1; i <= n; i++) visit[i] = false;
	queue<int> Q;
	Q.push(S), visit[S] = true;
	while (!Q.empty()) {
		int x = Q.front(); Q.pop();
		for (int y : adj[x]) {
			if (visit[y]) continue;
			visit[y] = true;
			Q.push(y);
		}
	}
}
int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);
	input();
	BFS(1);
	int ans = 0;
	for (int i = 2; i <= n; i++)
		if (visit[i])
			ans++;
	cout << ans;
	return 0;
}